### PR TITLE
Feat/deepseek chat node

### PR DESCRIPTION
**Pull request by nguyenhung10012003**
This pull request introduces support for DeepseekAI's API and models within the project. The changes include adding a new credential class for DeepseekAI, updating the models configuration, and creating a new node class for interacting with DeepseekAI's chat models.

### New Feature: DeepseekAI Integration

* **Credential Class for DeepseekAI**:
  - Added a new credential class `DeepseekApi` for handling DeepseekAI API keys. (`packages/components/credentials/DeepseekApi.credential.ts`)

* **Models Configuration**:
  - Updated the `models.json` file to include DeepseekAI models (`deepseek-chat` and `deepseek-coder`). (`packages/components/models.json`)

* **Node Class for DeepseekAI Chat Models**:
  - Created a new node class `Deepseek_ChatModels` to interact with DeepseekAI's chat models. This class includes various parameters such as model name, temperature, max tokens, and more. (`packages/components/nodes/chatmodels/Deepseek/Deepseek.ts`)

### CODE REVIEW

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                label: 'Allow Image Uploads',
+                name: 'allowImageUploads',
+                type: 'boolean',
+                description:
+                    'Allow image input. Refer to the <a href="https://docs.flowiseai.com/using-flowise/uploads#image" target="_blank">docs</a> for more details.',
+                default: false,
+                optional: true
+            },
+            {
+                label: 'Image Resolution',
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:133]: i dont think its capable of image input yet, we can remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            },
+            {
+                label: 'BaseOptions',
+                name: 'baseOptions',
+                type: 'json',
+                optional: true,
+                additionalParams: true
+            },
+            {
+                label: 'Allow Image Uploads',
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:124]: i dont think its capable of image input yet, we can remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        const maxTokens = nodeData.inputs?.maxTokens as string
+        const topP = nodeData.inputs?.topP as string
+        const frequencyPenalty = nodeData.inputs?.frequencyPenalty as string
+        const presencePenalty = nodeData.inputs?.presencePenalty as string
+        const timeout = nodeData.inputs?.timeout as string
+        const stopSequence = nodeData.inputs?.stopSequence as string
+        const streaming = nodeData.inputs?.streaming as boolean
+        const baseOptions = nodeData.inputs?.baseOptions
+
+        const allowImageUploads = nodeData.inputs?.allowImageUploads as boolean
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:177]: i dont think its capable of image input yet, we can remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        const imageResolution = nodeData.inputs?.imageResolution as string
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:178]: same here

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+        if (baseOptions) {
+            try {
+                parsedBaseOptions = typeof baseOptions === 'object' ? baseOptions : JSON.parse(baseOptions)
+            } catch (exception) {
+                throw new Error("Invalid JSON in the ChatOpenAI's BaseOptions: " + exception)
+            }
+        }
+
+        const multiModalOption: IMultiModalOption = {
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:218]: i dont think its capable of image input yet, we can remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+            image: {
+                allowImageUploads: allowImageUploads ?? false,
+                imageResolution
+            }
+        }
+
+        const model = new ChatOpenAI({
+            ...obj,
+            configuration: {
+                baseURL: this.baseURL
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:228]: missing `baseOptions: parsedBaseOptions`

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts (hunk 2 above)
Comment (nguyenhung10012003) [reply packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:124]: You're right, Deepseek currently does not support image uploads. I'm going to remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts (hunk 1 above)
Comment (nguyenhung10012003) [reply packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:133]: I'm going to remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts (hunk 3 above)
Comment (nguyenhung10012003) [reply packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:177]: Yes, I'm going to remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts (hunk 5 above)
Comment (nguyenhung10012003) [reply packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:218]: I'm going to remove this

File: packages/components/nodes/chatmodels/Deepseek/Deepseek.ts (hunk 6 above)
Comment (nguyenhung10012003) [reply packages/components/nodes/chatmodels/Deepseek/Deepseek.ts:228]: Sorry, that my fault, I've missed it

**Review by HenryHengZJ — APPROVED** [review]
looks good! thank you @nguyenhung10012003 !

**Pull request merged**